perception/map_based_prediction/src/map_based_prediction_node.cpp

```diff
@@ -1076,23 +1091,42 @@ LaneletsData MapBasedPredictionNode::getCurrentLanelets(const TrackedObject & ob
     return {};
   }
 
-  LaneletsData closest_lanelets;
+  LaneletsData object_lanelets;
+  std::optional<std::pair<double, lanelet::Lanelet>> closest_lanelet{std::nullopt};
   for (const auto & lanelet : surrounding_lanelets) {
     // Check if the close lanelets meet the necessary condition for start lanelets and
-    // Check if similar lanelet is inside the closest lanelet
+    // Check if similar lanelet is inside the object lanelet
     if (
       !checkCloseLaneletCondition(lanelet, object, search_point) ||
-      isDuplicated(lanelet, closest_lanelets)) {
+      isDuplicated(lanelet, object_lanelets)) {
       continue;
     }
 
-    LaneletData closest_lanelet;
-    closest_lanelet.lanelet = lanelet.second;
-    closest_lanelet.probability = calculateLocalLikelihood(lanelet.second, object);
-    closest_lanelets.push_back(closest_lanelet);
+    // Memorize closest lanelet
+    // NOTE: The object may be outside the lanelet.
+    if (
+      lanelet.first < dist_threshold_for_searching_lanelet_ &&
+      (!closest_lanelet || lanelet.first < closest_lanelet->first)) {
+      closest_lanelet = lanelet;
+    }
+
+    // Check if the obstacle is inside of this lanelet
+    constexpr double epsilon = 1e-3;
+    if (lanelet.first < epsilon) {
+      const auto object_lanelet =
+        LaneletData{lanelet.second, calculateLocalLikelihood(lanelet.second, object)};
+      object_lanelets.push_back(object_lanelet);
+    }
   }
 
-  return closest_lanelets;
+  if (!object_lanelets.empty()) {
+    return object_lanelets;
+  }
+  if (closest_lanelet) {
+    return LaneletsData{LaneletData{
+      closest_lanelet->second, calculateLocalLikelihood(closest_lanelet->second, object)}};
```

Comment:
need to check if `calculateLocalLikelihood` function is valid with obj outside the lanalet